controllers/hetznercluster_controller.go

```diff
@@ -500,73 +505,101 @@ func reconcileTargetSecret(ctx context.Context, clusterScope *scope.ClusterScope
+	// workload cluster client
+	wlClient, err := client.New(restConfig, client.Options{})
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to get client: %w", err)
 	}
 
-	secret := &corev1.Secret{
+	// To ensure compatibilty with both CCMs, create always a secret with name "hcloud" in the
+	// wl-cluster.
+	names := []string{clusterScope.HetznerCluster.Spec.HetznerSecret.Name}
+	if clusterScope.HetznerCluster.Spec.HetznerSecret.Name != "hcloud" {
+		names = append(names, "hcloud")
+	}
+
+	for _, name := range names {
+		err = reconcileOneWorkloadClusterSecret(ctx, clusterScope, wlClient, name)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to reconcile wl-cluster secret %q: %w",
+				name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func reconcileOneWorkloadClusterSecret(ctx context.Context, clusterScope *scope.ClusterScope, wlClient client.Client, name string) error {
+	wlSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      clusterScope.HetznerCluster.Spec.HetznerSecret.Name,
+			Name:      name,
 			Namespace: metav1.NamespaceSystem,
 		},
 	}
 
 	// Make sure secret exists and has the expected values
-	_, err = controllerutil.CreateOrUpdate(ctx, client, secret, func() error {
-		tokenSecretName := types.NamespacedName{
+	_, err := controllerutil.CreateOrUpdate(ctx, wlClient, wlSecret, func() error {
+		mgtSecretName := types.NamespacedName{
 			Namespace: clusterScope.HetznerCluster.Namespace,
 			Name:      clusterScope.HetznerCluster.Spec.HetznerSecret.Name,
 		}
 		secretManager := secretutil.NewSecretManager(clusterScope.Logger, clusterScope.Client, clusterScope.APIReader)
-		tokenSecret, err := secretManager.AcquireSecret(ctx, tokenSecretName, clusterScope.HetznerCluster, false, clusterScope.HetznerCluster.DeletionTimestamp.IsZero())
+		mgtSecret, err := secretManager.AcquireSecret(ctx, mgtSecretName, clusterScope.HetznerCluster, false, clusterScope.HetznerCluster.DeletionTimestamp.IsZero())
 		if err != nil {
 			return fmt.Errorf("failed to acquire secret: %w", err)
 		}
 
-		hetznerToken, keyExists := tokenSecret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken]
+		hcloudToken, keyExists := mgtSecret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken]
 		if !keyExists {
 			return fmt.Errorf("error key %s does not exist in secret/%s: %w",
 				clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken,
-				tokenSecretName,
+				mgtSecretName,
 				err,
 			)
 		}
 
-		if secret.Data == nil {
-			secret.Data = make(map[string][]byte)
+		if wlSecret.Data == nil {
+			wlSecret.Data = make(map[string][]byte)
 		}
 
-		secret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken] = hetznerToken
+		wlSecret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken] = hcloudToken
+
+		// upstream hcloud-ccm uses the secret key "token", while the old Syself ccm used "hcloud".
```

Comment:
can you specify what "used" means? This is configured in the Helm charts right?

Reply:
If I understand correctly, then "token" is the default for a configurable variable for the hcloud-ccm. If that is correct, then I would specify that here. Right now it sounds as if it is hardcoded

Reply:
done.

New text:

```go 
		// upstream hcloud-ccm uses by default the secret key "token", while the old Syself ccm used
		// "hcloud". For compatibility, we create always the other key, too.
```